keras/utils/data_utils.py

```diff
@@ -171,7 +171,7 @@ def get_file(fname,
         Path to the downloaded file
     """
     if cache_dir is None:
-        cache_dir = os.path.expanduser(os.path.join('~', '.keras'))
+        cache_dir = os.path.join('~', '.keras')
```

Comment:
I believe this should be `os.path.join(os.path.expanduser('~'), '.keras')`

Reply:
From the description of expanduser and join in the python doc, I believe that the following three pieces of code are equivalent:

```
# code in master
if cache_dir is None:
    cache_dir = os.path.expanduser(os.path.join('~', '.keras'))
...
datadir_base = os.path.expanduser(cache_dir)
```
```
# code that you are proposing
if cache_dir is None:
    cache_dir = os.path.join(os.path.expanduser('~'),  '.keras')
...
datadir_base = os.path.expanduser(cache_dir)
```

```
# code in PR
if cache_dir is None:
    cache_dir = os.path.join('~', '.keras')
...
datadir_base = os.path.expanduser(cache_dir)
```

But this is a small refactoring, if you think it is unnecessary, I understand.

Reply:
I don't doubt they're equivalent in unix. But the syntax we're using elsewhere (`os.path.join(os.path.expanduser('~'), '.keras')`) arose as a result of issues we had initially with Windows. 

Basically we started out with `os.path.join('~', '.keras')` and then moved to `os.path.join(os.path.expanduser('~'), '.keras')` to fix issues. See also: https://stackoverflow.com/questions/4028904/how-to-get-the-home-directory-in-python

Reply:
Thanks for the explanation, I'll change it.